bin/compile

```diff
@@ -134,6 +134,41 @@ else
   CACHED_PYTHON_STACK=$STACK
 fi
 
+# TODO: Move this into a new package manager handling implementation when adding Poetry support.
+if [[ ! -f requirements.txt && ! -f Pipfile && ! -f setup.py ]]; then
+  puts-warn
+  puts-warn "Error: No supported Python package manager files were found."
+  puts-warn
+  # We intentionally don't mention `setup.py` here since it's being removed soon.
+  puts-warn "A Python app on Heroku must have either a 'requirements.txt' or"
+  puts-warn "'Pipfile' file in the root directory of its source code, so the"
+  puts-warn "buildpack knows which dependencies to install."
+  puts-warn
+  puts-warn "Currently the root directory of your app contains:"
+  puts-warn
+  # TODO: Overhaul logging helpers so they can handle prefixing multi-line strings, and switch to them.
+  # shellcheck disable=SC2012 # Using `ls` instead of `find` is absolutely fine for this use case.
+  ls -1 --indicator-style=slash "${BUILD_DIR}" | sed 's/^/ !     /'
+  puts-warn
+  puts-warn "If you believe your app already has a 'requirements.txt' or"
+  puts-warn "'Pipfile' file, check that:"
+  puts-warn
+  puts-warn "1. The file is in the top level directory (not a subdirectory)."
+  puts-warn "2. The filename has the correct spelling and capitalisation."
```

Comment:
Not really an issue but something I'm curious about - do we prefer `en-US` spelling over other locales where this would be `capitalization` instead of `capitalisation`?  I doubt most folks would notice this but, as someone who's done localization work and as a fellow non-American, inconsistencies like this stand out to me 😅

Reply:
At Mozilla en-US vs en-GB was document-author/component-owners call. I don't know about here :-)